src/EnergyPlus/MixedAir.cc

```diff
@@ -4593,17 +4718,34 @@ namespace MixedAir {
 
 		if ( OAController( OAControllerNum ).ControllerType_Num == ControllerOutsideAir ) {
 			// The outside air controller sets the outside air flow rate and the relief air flow rate
-			Node( OutAirNodeNum ).MassFlowRate = OAController( OAControllerNum ).OAMassFlow;
-			Node( InletAirNodeNum ).MassFlowRate = OAController( OAControllerNum ).OAMassFlow;
-			Node( RelAirNodeNum ).MassFlowRate = OAController( OAControllerNum ).RelMassFlow;
-			Node( OutAirNodeNum ).MassFlowRateMaxAvail = OAController( OAControllerNum ).OAMassFlow;
+			if (OAController(OAControllerNum).ManageDemand)
+			{
+				Node(OutAirNodeNum).MassFlowRate = OAController(OAControllerNum).DemandLimitFlowRate;
+				Node(InletAirNodeNum).MassFlowRate = OAController(OAControllerNum).DemandLimitFlowRate;
+				Node(OutAirNodeNum).MassFlowRateMaxAvail = OAController(OAControllerNum).DemandLimitFlowRate;
+			}
+			else
+			{
+				Node(OutAirNodeNum).MassFlowRate = OAController(OAControllerNum).OAMassFlow;
+				Node(InletAirNodeNum).MassFlowRate = OAController(OAControllerNum).OAMassFlow;
```

Comment:
I didn't check the relationship between these routines. Although it is in a different routine,  "OAController( OAControllerNum ).OAMassFlow" may still have the same value as  "OAController(OAControllerNum).DemandLimitFlowRate" since it was set at Line 4272. Please have check. If it is not set as "OAController(OAControllerNum).DemandLimitFlowRate" here, we can keep your codes here.